2-mandatory/4-stocks.js

```diff
@@ -21,53 +22,84 @@ const CLOSING_PRICES_LAST_5_DAYS_FOR_ALL_STOCKS = [
 /*
     We want to understand what the average price over the last 5 days for each stock is.
     Implement the below function, which
-        - Takes this CLOSING_PRICES_LAST_5_DAYS_FOR_ALL_STOCKS array as input (remember, it's an array of arrays)
+        - Takes this CLOSING_PRICES_LAST_5_DAYS_FOR_ALL_STOCKS array as input
+         (remember, it's an array of arrays)
         - Returns an array containing the average price over the last 5 days for each stock. 
-            For example, the first element of the resulting array should contain Apple’s (aapl) average stock price for the last 5 days.
+            For example, the first element of the resulting array should contain 
+            Apple’s (aapl) average stock price for the last 5 days.
             The second element should be Microsoft's (msft) average price, and so on.
     The average value should be rounded to 2 decimal places, and should be a number (not a string)
 
-    Hint 1: To calculate the average of a set of values, you can add them together and divide by the number of values.
+    Hint 1: To calculate the average of a set of values, you can add them together and
+     divide by the number of values.
         So the average of 5, 10 and 20 is (5 + 10 + 20) / 3 = 11.67
     Hint 2: If the problem seems complex, try breaking it down into smaller problems.
         Solve the smaller problems, and then build those solutions back up to solve the larger problem.
         Functions can help with this!
 */
 function getAveragePrices(closingPricesForAllStocks) {
     // TODO
+    let stockPriceTotal = 0;
+    let stockPriceAverage = [];
+    for(let i=0 ; i < closingPricesForAllStocks.length; i++){
+        let stockPrice = closingPricesForAllStocks[i];
+            for (let j = 0; j < stockPrice.length; j++){
+                stockPriceTotal+= stockPrice[j];
+        }
+        stockPriceAverage.push(number((stockPriceTotal / stockPrice.length).toFixed(2)));
```

Comment:
I don't think your method of rounding is working, check out this way: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/round

Reply:
See how they use it for two decimal places here: https://stackoverflow.com/questions/11832914/how-to-round-to-at-most-2-decimal-places-if-necessary